app/controllers/admin/api/services/proxy/configs_controller.rb

```diff
@@ -106,12 +108,14 @@ def proxy_config
   end
 
   def service_proxy_configs
-    service.proxy.proxy_configs.by_environment(environment)
+    ProxyConfig
+      .by_environment(environment)
+      .where(proxy_id: ProxiesForServiceOwnerAndWatcherQuery.call(owner: service, watcher: current_account).select(:id))
```

Comment:
I think it would be more idiomatic with a simple `service.proxy_configs`. Defining a class to delegate the the filtering of services that are permitted to the current user breaks the way we doing in other controllers.